dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageCheckOption.java

```diff
@@ -105,45 +88,35 @@ static StaticLinkageCheckOption parseArguments(String[] arguments) throws ParseE
         "report-only-reachable",
         false,
         "To report only linkage errors reachable from entry point");
+    return options;
+  }
 
-    CommandLineParser parser = new DefaultParser();
-    List<Path> jarFilePaths = new ArrayList<>();
-
-    ImmutableList.Builder<String> mavenCoordinates = ImmutableList.builder();
-    try {
-      CommandLine commandLine = parser.parse(options, arguments);
-      if (Stream.of('b', 'a', 'j').filter(commandLine::hasOption).count() > 1) {
-        throw new IllegalArgumentException(
-            "One of BOM, Maven coordinates, or jar files can be specified");
+  static ImmutableList<Path> generateInputClasspath(CommandLine commandLine)
+      throws RepositoryException, ParseException {
+    
+    ImmutableList.Builder<Path> jarFileBuilder = ImmutableList.builder();
+    
+    Splitter commaSplitter = Splitter.on(",");
+    if (commandLine.hasOption("a")) {
+      String mavenCoordinatesOption = commandLine.getOptionValue("a");
+      for (String coord : commaSplitter.split(mavenCoordinatesOption)) {
+        jarFileBuilder.addAll(StaticLinkageChecker.coordinatesToClasspath(coord));
       }
-      Splitter commaSplitter = Splitter.on(",");
-      if (commandLine.hasOption("a")) {
-        String mavenCoordinatesOption = commandLine.getOptionValue("a");
-        mavenCoordinates.addAll(commaSplitter.split(mavenCoordinatesOption));
-      }
-      if (commandLine.hasOption("j")) {
-        String jarFiles = commandLine.getOptionValue("j");
-        List<Path> jarFilesInArguments =
-            Streams.stream(commaSplitter.split(jarFiles))
-                .map(name -> (Paths.get(name)).toAbsolutePath())
-                .collect(Collectors.toList());
-        jarFilePaths.addAll(jarFilesInArguments);
-      }
-
+      return jarFileBuilder.build();
+    } else if (commandLine.hasOption("j")) {
+      String jarFiles = commandLine.getOptionValue("j");
+      List<Path> jarFilesInArguments =
+          Streams.stream(commaSplitter.split(jarFiles))
+              .map(name -> (Paths.get(name)).toAbsolutePath())
+              .collect(Collectors.toList());
```

Comment:
`ImmutableList.toImmutableList()` lets you collect directly into an `ImmutableList`.